tests/compiler/test_list.py

```diff
@@ -556,3 +556,14 @@ def test_relation_list_duplicate_assignment(snippetcompiler):
         )
     )
     compiler.do_compile()
+
+
+def test_error_list_validation(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+        std::print(std::count("hello"))
+        """
+    )
+    with pytest.raises(RuntimeException) as exc_info:
+        (_, scopes) = compiler.do_compile()
+    assert "Invalid value 'hello', expected list" in exc_info.value.msg
```

Comment:
`pytest.raises` has a `match` kwarg for this. Just an FYI, not a change request.